src/charm.py

```diff
@@ -204,50 +216,89 @@ def _on_clean_up_packages_action(self, event):
         start = time.time()
         logger.info("Cleaning up unreferenced packages")
         packages_to_be_removed, freed_up_space = self._check_packages()
-        prefix_message = "Clean up completed without errors."
-        for package in packages_to_be_removed:
-            try:
-                package.unlink()
-                logger.info("Removed {}".format(package))
-            except Exception as e:
-                logger.error(e)
-                prefix_message = (
-                    "Clean up completed with errors, "
-                    "please refer to juju's log for details."
-                )
+        cleanup_result = clean_packages(packages_to_be_removed)
         elapsed = time.time() - start
-        logger.info("Clean up complete, took {}s".format(elapsed))
 
-        event.set_results(
-            {
-                "message": "{} Freed up {}".format(prefix_message, freed_up_space),
-            }
-        )
+        if cleanup_result is True:
+            message = "Clean up completed without errors."
+        else:
+            message = (
+                "Clean up completed with errors, please refer to juju's log for "
+                "details."
+            )
+
+        logger.info("Clean up complete, took %ds", elapsed)
+        event.set_results({"message": f"{message} Freed up {freed_up_space}"})
```

Comment:
For consistency, it would be better to add a package count here as well
```suggestion
        event.set_results(
            {
                "time": elapsed,
                "message": "{} Freed up {} by cleaning {} packages".format(
                    message, freed_up_space, len(packages_to_be_removed)
            }
        )
```